### DISCUSSION TITLE
How to save the time column in the solution log to a variable？

### DISCUSSION THREAD

**Discussion by xiaoyezi55** [google/or-tools#4117] [Linear Solver questions] [answered]
solver = pywraplp.Solver.CreateSolver('SCIP')

solver.EnableOutput()

How to save the time column in the solution log to a variable？

![1111188888888](https://github.com/google/or-tools/assets/32790564/c98bcef3-2441-4170-9c14-e243749c342d)

**Comment by lperron** [comment]
scip outputs to stdout. You need to intercept this.

Math-opt will allow to intercept this with a callback.

**Reply by xiaoyezi55** [reply]
May I ask how to implement it

**Comment by lperron** [comment]
I do not know. This is a python question, not an OR-Tools one.
Laurent Perron | Operations Research | ***@***.*** | (33) 1 42 68 53
00

**Reply by xiaoyezi55** [reply]
Is there any way to obtain the OR-Tools optimization time progress percentage？

**Comment by lperron** [comment, marked as answer]
the question makes no sense.

there is not a 'ortools optimization time progress'.

You create the time limit. Why don't you check time in python and compare
it with the time limit you set ?
Laurent Perron | Operations Research | ***@***.*** | (33) 1 42 68 53
00

**Reply by xiaoyezi55** [reply]
import time
from ortools.linear_solver import pywraplp

# 创建求解器并设置时间限制
solver = pywraplp.Solver.CreateSolver('SCIP')
solver.set_time_limit(300000)  # 设置时间限制为300秒

# 定义你的优化问题
# ...

# 记录开始时间
start_time = time.time()

# 求解问题
status = solver.Solve()

if status == pywraplp.Solver.OPTIMAL or status == pywraplp.Solver.FEASIBLE:
    end_time = time.time()
    elapsed_time = end_time - start_time
    progress_percentage = 100  
    print(f"Optimization completed in {elapsed_time} seconds. Progress: {progress_percentage}%")
else:
 
    end_time = time.time()
    elapsed_time = end_time - start_time
    progress_percentage = (elapsed_time / 300000) * 100  
    print(f"Optimization still running. Elapsed time: {elapsed_time} seconds. Estimated progress: {progress_percentage}%")



The above code can only output optimization progress after optimization is completed. How to output optimization time progress percentage during the optimization process

**Comment by lperron** [comment]
is your problem integral ?

If so, use CP-SAT. It is a better ILP solver, and it supports solution
callbacks in all languages.
Laurent Perron | Operations Research | ***@***.*** | (33) 1 42 68 53
00

**Reply by xiaoyezi55** [reply]
OK thanks!
